### PR TITLE
Fixes #7992: Openssl build breaks debian packaging

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7992

### CODE REVIEW

File: rudder-agent/SOURCES/Makefile [hunk 1]
```diff
@@ -45,14 +45,12 @@ FETCH = fetch -q -o
 ifneq (,$(wildcard /usr/bin/curl))
 GET = $(PROXY_ENV) $(CURL)
 GET_OPT = --curl
-else
-ifneq (,$(wildcard /usr/bin/fetch))
+else ifneq (,$(wildcard /usr/bin/fetch))
```
Comment (peckpeck) [inline rudder-agent/SOURCES/Makefile]: This is just the correct way to write elsif in makefiles

File: rudder-agent/SOURCES/Makefile [hunk 2]
```diff
@@ -90,6 +88,9 @@ endif
 # Hack needed because openssl is not guessing correctly the architecture on solaris 32bits
 ifeq (i386-pc-solaris2.10,$(shell ./config.guess))
 SSL_CONFIGURE:=./Configure solaris-x86-gcc
+# Hack needed because openssl is not guessing correctly the architecture on debian 32 bits chroot
+else ifeq (i386,$(shell dpkg --print-architecture))
```
Comment (jooooooon) [inline rudder-agent/SOURCES/Makefile]: This won't work on non-debian systems.

Comment (jooooooon) [inline rudder-agent/SOURCES/Makefile]: I propose to check the dpkg command is available before running it.

**Comment by peckpeck** [comment]
PR updated

File: rudder-agent/SOURCES/Makefile [hunk 3]
```diff
@@ -90,6 +88,9 @@ endif
 # Hack needed because openssl is not guessing correctly the architecture on solaris 32bits
 ifeq (i386-pc-solaris2.10,$(shell ./config.guess))
 SSL_CONFIGURE:=./Configure solaris-x86-gcc
+# Hack needed because openssl is not guessing correctly the architecture on debian 32 bits chroot
+else ifeq (i386,$(shell [ -x /usr/bin/dpkg ] && dpkg --print-architecture))
```
Comment (jooooooon) [inline rudder-agent/SOURCES/Makefile]: If /usr/bin/dpkg does not exist, this will return 1. Will make complain?

Comment (peckpeck) [inline rudder-agent/SOURCES/Makefile]: No, it will just ignore it

**Pull request merged**